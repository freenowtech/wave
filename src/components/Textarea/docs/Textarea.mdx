---
name: Textarea
menu: Components
route: /components/textarea
---

import { Playground } from 'docz';
import { ControlledTextarea } from './ControlledTextarea';
import { Textarea } from '../Textarea';
import { ItemWrapper } from '../../../../docs/components/ItemWrapper';
import { StyledSystemLinks } from '../../../docs/StyledSystemLinks';
import { TextareaPropsTable } from './TextareaPropsTable';


# Textarea

<StyledSystemLinks component="Textarea" supportedProps={ ['margin', 'height', 'width'] }/>

## Properties

<TextareaPropsTable />

## Playground

<Playground>
    <Textarea placeholder="user@free-now.com" label="E-Mail Address" resize="vertical" />
</Playground>

## Appearance

### Default

<Textarea label="Default" />

### Default placeholder

<Textarea label="Default" placeholder="Placeholder" />

### Error

<Textarea label="Default" error />

### Disabled

<Textarea label="Default" disabled />

### Inverted

<ItemWrapper inverted gridTemplate="60% 1fr" mt={4}>
    <Textarea inverted label="Default" />
</ItemWrapper>

