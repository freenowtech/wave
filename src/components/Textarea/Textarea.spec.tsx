import { render, screen } from '@testing-library/react';
import * as React from 'react';
import { Textarea } from './Textarea';

describe('Textarea', () => {
    it('renders', () => {
        expect(render(<Textarea />).container.firstChild).toMatchSnapshot();
    });

    it('renders the inverted style', () => {
        expect(render(<Textarea inverted />).container.firstChild).toMatchSnapshot();
    });

    it('renders the label', () => {
        expect(render(<Textarea label="Name" />).container.firstChild).toMatchSnapshot();
    });

    it('renders the label and the placeholder', () => {
        expect(render(<Textarea label="Name" placeholder="FREE NOW" />).container.firstChild).toMatchSnapshot();
    });

    it('renders the error state', () => {
        expect(render(<Textarea error />).container.firstChild).toMatchSnapshot();
    });

    it('renders the error state with label and placeholder', () => {
        expect(render(<Textarea label="Name" placeholder="FREE NOW" error />).container.firstChild).toMatchSnapshot();
    });

    it('renders with custom height and width', () => {
        expect(
            render(<Textarea label="Name" placeholder="FREE NOW" height="10rem" width="10rem" />).container.firstChild
        ).toMatchSnapshot();
    });

    it('renders the disabled state with label and placeholder', () => {
        expect(
            render(<Textarea label="Name" placeholder="FREE NOW" disabled />).container.firstChild
        ).toMatchSnapshot();
    });

    describe('link textarea with the label', () => {
        it('uses `id` prop value if passed', () => {
            render(<Textarea id="test-input-id" label="Simple Label" />);

            expect(screen.getByLabelText('Simple Label')).toHaveAttribute('id', 'test-input-id');
            expect(screen.getByText('Simple Label')).toHaveAttribute('for', 'test-input-id');
        });

        it('generate id automatically if `id` prop is empty', () => {
            render(<Textarea label="Simple Label" />);
            const generatedId = 'random';

            expect(screen.getByLabelText('Simple Label')).toHaveAttribute('id', generatedId);
            expect(screen.getByText('Simple Label')).toHaveAttribute('for', generatedId);
        });
    });
});
