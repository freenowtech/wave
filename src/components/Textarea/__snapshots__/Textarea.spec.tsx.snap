// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea renders 1`] = `
.c0 {
  display: inline-block;
  position: relative;
  box-sizing: border-box;
  width: 21.5rem;
  height: 7.25rem;
}

.c1 {
  margin: 0;
  box-sizing: border-box;
  background: #FFFFFF;
  border-radius: 0;
  color: #001E3E;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  -webkit-transition: box-shadow 100ms,border 100ms;
  transition: box-shadow 100ms,border 100ms;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 100%;
  border-radius: 0.25rem;
  border: 0.0625rem solid #C6CDD4;
  font-size: 1rem;
  height: 3rem;
  padding: 0 0.75rem;
  height: 100%;
  padding: 0.75rem;
  overflow: auto;
  resize: both;
}

.c1::-webkit-input-placeholder {
  color: #9CA7B4;
}

.c1::-moz-placeholder {
  color: #9CA7B4;
}

.c1:-ms-input-placeholder {
  color: #9CA7B4;
}

.c1::placeholder {
  color: #9CA7B4;
}

.c1:active,
.c1:focus {
  border-color: #096BDB;
  box-shadow: inset 0 0 0 0.0625rem #096BDB;
}

.c1:disabled {
  color: #C6CDD4;
  border-color: #C6CDD4;
  box-shadow: none;
  cursor: not-allowed;
}

.c1:disabled::-webkit-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::-moz-placeholder {
  color: #C6CDD4;
}

.c1:disabled:-ms-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::placeholder {
  color: #C6CDD4;
}

.c1:-webkit-autofill,
.c1:-webkit-autofill:hover,
.c1:-webkit-autofill:focus,
.c1:-webkit-autofill:active {
  -webkit-text-fill-color: #001E3E;
  -webkit-transition: background-color 99999999ms ease 99999999ms;
  transition: background-color 99999999ms ease 99999999ms;
}

.c1 + .sc-AxiKw {
  color: #9CA7B4;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

.c1:disabled + .sc-AxiKw {
  color: #C6CDD4;
}

.c1:-webkit-autofill + .sc-AxiKw,
.c1:-webkit-autofill:hover + .sc-AxiKw,
.c1:-webkit-autofill:focus + .sc-AxiKw,
.c1:-webkit-autofill:active + .sc-AxiKw {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
}

.c1:focus:not(:disabled) + .sc-AxiKw {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
  color: #096BDB;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

<div
  class="c0"
>
  <textarea
    class="sc-AxjAm sc-AxhCb c1"
  />
</div>
`;

exports[`Textarea renders the disabled state with label and placeholder 1`] = `
.c3 {
  position: absolute;
  pointer-events: none;
  background-color: transparent;
  line-height: 1.5;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  max-width: calc(100% - 2rem);
  -webkit-transition: top 100ms ease-out,left 100ms ease-out, padding 100ms ease-out,font-size 100ms ease-out, color 100ms ease-out,background 100ms ease-out;
  transition: top 100ms ease-out,left 100ms ease-out, padding 100ms ease-out,font-size 100ms ease-out, color 100ms ease-out,background 100ms ease-out;
  top: 0.75rem;
  left: 0.5rem;
  padding: 0 0.25rem;
  font-size: 1rem;
}

.c0 {
  display: inline-block;
  position: relative;
  box-sizing: border-box;
  width: 21.5rem;
  height: 7.25rem;
}

.c1 {
  margin: 0;
  box-sizing: border-box;
  background: #FFFFFF;
  border-radius: 0;
  color: #001E3E;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  -webkit-transition: box-shadow 100ms,border 100ms;
  transition: box-shadow 100ms,border 100ms;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 100%;
  border-radius: 0.25rem;
  border: 0.0625rem solid #C6CDD4;
  font-size: 1rem;
  height: 3rem;
  padding: 0 0.75rem;
  height: 100%;
  padding: 0.75rem;
  overflow: auto;
  resize: both;
}

.c1::-webkit-input-placeholder {
  color: #9CA7B4;
}

.c1::-moz-placeholder {
  color: #9CA7B4;
}

.c1:-ms-input-placeholder {
  color: #9CA7B4;
}

.c1::placeholder {
  color: #9CA7B4;
}

.c1:active,
.c1:focus {
  border-color: #096BDB;
  box-shadow: inset 0 0 0 0.0625rem #096BDB;
}

.c1:disabled {
  color: #C6CDD4;
  border-color: #C6CDD4;
  box-shadow: none;
  cursor: not-allowed;
}

.c1:disabled::-webkit-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::-moz-placeholder {
  color: #C6CDD4;
}

.c1:disabled:-ms-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::placeholder {
  color: #C6CDD4;
}

.c1:-webkit-autofill,
.c1:-webkit-autofill:hover,
.c1:-webkit-autofill:focus,
.c1:-webkit-autofill:active {
  -webkit-text-fill-color: #001E3E;
  -webkit-transition: background-color 99999999ms ease 99999999ms;
  transition: background-color 99999999ms ease 99999999ms;
}

.c1 + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
  color: #9CA7B4;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

.c1:disabled + .c2 {
  color: #C6CDD4;
}

.c1:-webkit-autofill + .c2,
.c1:-webkit-autofill:hover + .c2,
.c1:-webkit-autofill:focus + .c2,
.c1:-webkit-autofill:active + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
}

.c1:focus:not(:disabled) + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
  color: #096BDB;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

<div
  class="c0"
>
  <textarea
    class="sc-AxjAm sc-AxhCb c1"
    disabled=""
    placeholder="FREE NOW"
  />
  <label
    class="sc-AxirZ c2 c3"
  >
    Name
  </label>
</div>
`;

exports[`Textarea renders the error state 1`] = `
.c0 {
  display: inline-block;
  position: relative;
  box-sizing: border-box;
  width: 21.5rem;
  height: 7.25rem;
}

.c1 {
  margin: 0;
  box-sizing: border-box;
  background: #FFFFFF;
  border-radius: 0;
  color: #001E3E;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  -webkit-transition: box-shadow 100ms,border 100ms;
  transition: box-shadow 100ms,border 100ms;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 100%;
  border-radius: 0.25rem;
  border: 0.0625rem solid #C6CDD4;
  font-size: 1rem;
  height: 3rem;
  padding: 0 0.75rem;
  box-shadow: inset 0 0 0 0.0625rem #FF5E05;
  border-color: #FF5E05;
  height: 100%;
  padding: 0.75rem;
  overflow: auto;
  resize: both;
}

.c1::-webkit-input-placeholder {
  color: #9CA7B4;
}

.c1::-moz-placeholder {
  color: #9CA7B4;
}

.c1:-ms-input-placeholder {
  color: #9CA7B4;
}

.c1::placeholder {
  color: #9CA7B4;
}

.c1:active,
.c1:focus {
  border-color: #096BDB;
  box-shadow: inset 0 0 0 0.0625rem #096BDB;
}

.c1:disabled {
  color: #C6CDD4;
  border-color: #C6CDD4;
  box-shadow: none;
  cursor: not-allowed;
}

.c1:disabled::-webkit-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::-moz-placeholder {
  color: #C6CDD4;
}

.c1:disabled:-ms-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::placeholder {
  color: #C6CDD4;
}

.c1:-webkit-autofill,
.c1:-webkit-autofill:hover,
.c1:-webkit-autofill:focus,
.c1:-webkit-autofill:active {
  -webkit-text-fill-color: #001E3E;
  -webkit-transition: background-color 99999999ms ease 99999999ms;
  transition: background-color 99999999ms ease 99999999ms;
}

.c1 + .sc-AxiKw {
  color: #9CA7B4;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

.c1 ~ .sc-AxiKw {
  color: #FF5E05;
}

.c1:disabled + .sc-AxiKw {
  color: #C6CDD4;
}

.c1:-webkit-autofill + .sc-AxiKw,
.c1:-webkit-autofill:hover + .sc-AxiKw,
.c1:-webkit-autofill:focus + .sc-AxiKw,
.c1:-webkit-autofill:active + .sc-AxiKw {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
}

.c1:focus:not(:disabled) + .sc-AxiKw {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
  color: #096BDB;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

<div
  class="c0"
>
  <textarea
    class="sc-AxjAm sc-AxhCb c1"
  />
</div>
`;

exports[`Textarea renders the error state with label and placeholder 1`] = `
.c3 {
  position: absolute;
  pointer-events: none;
  background-color: transparent;
  line-height: 1.5;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  max-width: calc(100% - 2rem);
  -webkit-transition: top 100ms ease-out,left 100ms ease-out, padding 100ms ease-out,font-size 100ms ease-out, color 100ms ease-out,background 100ms ease-out;
  transition: top 100ms ease-out,left 100ms ease-out, padding 100ms ease-out,font-size 100ms ease-out, color 100ms ease-out,background 100ms ease-out;
  top: 0.75rem;
  left: 0.5rem;
  padding: 0 0.25rem;
  font-size: 1rem;
}

.c0 {
  display: inline-block;
  position: relative;
  box-sizing: border-box;
  width: 21.5rem;
  height: 7.25rem;
}

.c1 {
  margin: 0;
  box-sizing: border-box;
  background: #FFFFFF;
  border-radius: 0;
  color: #001E3E;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  -webkit-transition: box-shadow 100ms,border 100ms;
  transition: box-shadow 100ms,border 100ms;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 100%;
  border-radius: 0.25rem;
  border: 0.0625rem solid #C6CDD4;
  font-size: 1rem;
  height: 3rem;
  padding: 0 0.75rem;
  box-shadow: inset 0 0 0 0.0625rem #FF5E05;
  border-color: #FF5E05;
  height: 100%;
  padding: 0.75rem;
  overflow: auto;
  resize: both;
}

.c1::-webkit-input-placeholder {
  color: #9CA7B4;
}

.c1::-moz-placeholder {
  color: #9CA7B4;
}

.c1:-ms-input-placeholder {
  color: #9CA7B4;
}

.c1::placeholder {
  color: #9CA7B4;
}

.c1:active,
.c1:focus {
  border-color: #096BDB;
  box-shadow: inset 0 0 0 0.0625rem #096BDB;
}

.c1:disabled {
  color: #C6CDD4;
  border-color: #C6CDD4;
  box-shadow: none;
  cursor: not-allowed;
}

.c1:disabled::-webkit-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::-moz-placeholder {
  color: #C6CDD4;
}

.c1:disabled:-ms-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::placeholder {
  color: #C6CDD4;
}

.c1:-webkit-autofill,
.c1:-webkit-autofill:hover,
.c1:-webkit-autofill:focus,
.c1:-webkit-autofill:active {
  -webkit-text-fill-color: #001E3E;
  -webkit-transition: background-color 99999999ms ease 99999999ms;
  transition: background-color 99999999ms ease 99999999ms;
}

.c1 + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
  color: #9CA7B4;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

.c1 ~ .c2 {
  color: #FF5E05;
}

.c1:disabled + .c2 {
  color: #C6CDD4;
}

.c1:-webkit-autofill + .c2,
.c1:-webkit-autofill:hover + .c2,
.c1:-webkit-autofill:focus + .c2,
.c1:-webkit-autofill:active + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
}

.c1:focus:not(:disabled) + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
  color: #096BDB;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

<div
  class="c0"
>
  <textarea
    class="sc-AxjAm sc-AxhCb c1"
    placeholder="FREE NOW"
  />
  <label
    class="sc-AxirZ c2 c3"
  >
    Name
  </label>
</div>
`;

exports[`Textarea renders the inverted style 1`] = `
.c0 {
  display: inline-block;
  position: relative;
  box-sizing: border-box;
  width: 21.5rem;
  height: 7.25rem;
}

.c1 {
  margin: 0;
  box-sizing: border-box;
  background: transparent;
  border-radius: 0;
  color: #FFFFFF;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  -webkit-transition: box-shadow 100ms,border 100ms;
  transition: box-shadow 100ms,border 100ms;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 100%;
  border-radius: 0.25rem;
  border: 0.0625rem solid #C6CDD4;
  font-size: 1rem;
  height: 3rem;
  padding: 0 0.75rem;
  height: 100%;
  padding: 0.75rem;
  overflow: auto;
  resize: both;
}

.c1::-webkit-input-placeholder {
  color: #C6CDD4;
}

.c1::-moz-placeholder {
  color: #C6CDD4;
}

.c1:-ms-input-placeholder {
  color: #C6CDD4;
}

.c1::placeholder {
  color: #C6CDD4;
}

.c1:active,
.c1:focus {
  border-color: #096BDB;
  box-shadow: inset 0 0 0 0.0625rem #096BDB;
}

.c1:disabled {
  color: #637689;
  border-color: #637689;
  box-shadow: none;
  cursor: not-allowed;
}

.c1:disabled::-webkit-input-placeholder {
  color: #637689;
}

.c1:disabled::-moz-placeholder {
  color: #637689;
}

.c1:disabled:-ms-input-placeholder {
  color: #637689;
}

.c1:disabled::placeholder {
  color: #637689;
}

.c1:-webkit-autofill,
.c1:-webkit-autofill:hover,
.c1:-webkit-autofill:focus,
.c1:-webkit-autofill:active {
  -webkit-text-fill-color: #FFFFFF;
  -webkit-transition: background-color 99999999ms ease 99999999ms;
  transition: background-color 99999999ms ease 99999999ms;
}

.c1 + .sc-AxiKw {
  color: #C6CDD4;
  background: #001E3E;
  background: linear-gradient(0deg,#001E3E calc(50% + 0.0625rem),transparent 50%);
}

.c1:disabled + .sc-AxiKw {
  color: #637689;
}

.c1:-webkit-autofill + .sc-AxiKw,
.c1:-webkit-autofill:hover + .sc-AxiKw,
.c1:-webkit-autofill:focus + .sc-AxiKw,
.c1:-webkit-autofill:active + .sc-AxiKw {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
}

.c1:focus:not(:disabled) + .sc-AxiKw {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
  color: #FFFFFF;
  background: #001E3E;
  background: linear-gradient(0deg,#001E3E calc(50% + 0.0625rem),transparent 50%);
}

<div
  class="c0"
>
  <textarea
    class="sc-AxjAm sc-AxhCb c1"
  />
</div>
`;

exports[`Textarea renders the label 1`] = `
.c3 {
  position: absolute;
  pointer-events: none;
  background-color: transparent;
  line-height: 1.5;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  max-width: calc(100% - 2rem);
  -webkit-transition: top 100ms ease-out,left 100ms ease-out, padding 100ms ease-out,font-size 100ms ease-out, color 100ms ease-out,background 100ms ease-out;
  transition: top 100ms ease-out,left 100ms ease-out, padding 100ms ease-out,font-size 100ms ease-out, color 100ms ease-out,background 100ms ease-out;
  top: 0.75rem;
  left: 0.5rem;
  padding: 0 0.25rem;
  font-size: 1rem;
}

.c0 {
  display: inline-block;
  position: relative;
  box-sizing: border-box;
  width: 21.5rem;
  height: 7.25rem;
}

.c1 {
  margin: 0;
  box-sizing: border-box;
  background: #FFFFFF;
  border-radius: 0;
  color: #001E3E;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  -webkit-transition: box-shadow 100ms,border 100ms;
  transition: box-shadow 100ms,border 100ms;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 100%;
  border-radius: 0.25rem;
  border: 0.0625rem solid #C6CDD4;
  font-size: 1rem;
  height: 3rem;
  padding: 0 0.75rem;
  height: 100%;
  padding: 0.75rem;
  overflow: auto;
  resize: both;
}

.c1::-webkit-input-placeholder {
  color: #9CA7B4;
}

.c1::-moz-placeholder {
  color: #9CA7B4;
}

.c1:-ms-input-placeholder {
  color: #9CA7B4;
}

.c1::placeholder {
  color: #9CA7B4;
}

.c1:active,
.c1:focus {
  border-color: #096BDB;
  box-shadow: inset 0 0 0 0.0625rem #096BDB;
}

.c1:disabled {
  color: #C6CDD4;
  border-color: #C6CDD4;
  box-shadow: none;
  cursor: not-allowed;
}

.c1:disabled::-webkit-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::-moz-placeholder {
  color: #C6CDD4;
}

.c1:disabled:-ms-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::placeholder {
  color: #C6CDD4;
}

.c1:-webkit-autofill,
.c1:-webkit-autofill:hover,
.c1:-webkit-autofill:focus,
.c1:-webkit-autofill:active {
  -webkit-text-fill-color: #001E3E;
  -webkit-transition: background-color 99999999ms ease 99999999ms;
  transition: background-color 99999999ms ease 99999999ms;
}

.c1 + .c2 {
  color: #9CA7B4;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

.c1:disabled + .c2 {
  color: #C6CDD4;
}

.c1:-webkit-autofill + .c2,
.c1:-webkit-autofill:hover + .c2,
.c1:-webkit-autofill:focus + .c2,
.c1:-webkit-autofill:active + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
}

.c1:focus:not(:disabled) + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
  color: #096BDB;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

<div
  class="c0"
>
  <textarea
    class="sc-AxjAm sc-AxhCb c1"
  />
  <label
    class="sc-AxirZ c2 c3"
  >
    Name
  </label>
</div>
`;

exports[`Textarea renders the label and the placeholder 1`] = `
.c3 {
  position: absolute;
  pointer-events: none;
  background-color: transparent;
  line-height: 1.5;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  max-width: calc(100% - 2rem);
  -webkit-transition: top 100ms ease-out,left 100ms ease-out, padding 100ms ease-out,font-size 100ms ease-out, color 100ms ease-out,background 100ms ease-out;
  transition: top 100ms ease-out,left 100ms ease-out, padding 100ms ease-out,font-size 100ms ease-out, color 100ms ease-out,background 100ms ease-out;
  top: 0.75rem;
  left: 0.5rem;
  padding: 0 0.25rem;
  font-size: 1rem;
}

.c0 {
  display: inline-block;
  position: relative;
  box-sizing: border-box;
  width: 21.5rem;
  height: 7.25rem;
}

.c1 {
  margin: 0;
  box-sizing: border-box;
  background: #FFFFFF;
  border-radius: 0;
  color: #001E3E;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  -webkit-transition: box-shadow 100ms,border 100ms;
  transition: box-shadow 100ms,border 100ms;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 100%;
  border-radius: 0.25rem;
  border: 0.0625rem solid #C6CDD4;
  font-size: 1rem;
  height: 3rem;
  padding: 0 0.75rem;
  height: 100%;
  padding: 0.75rem;
  overflow: auto;
  resize: both;
}

.c1::-webkit-input-placeholder {
  color: #9CA7B4;
}

.c1::-moz-placeholder {
  color: #9CA7B4;
}

.c1:-ms-input-placeholder {
  color: #9CA7B4;
}

.c1::placeholder {
  color: #9CA7B4;
}

.c1:active,
.c1:focus {
  border-color: #096BDB;
  box-shadow: inset 0 0 0 0.0625rem #096BDB;
}

.c1:disabled {
  color: #C6CDD4;
  border-color: #C6CDD4;
  box-shadow: none;
  cursor: not-allowed;
}

.c1:disabled::-webkit-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::-moz-placeholder {
  color: #C6CDD4;
}

.c1:disabled:-ms-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::placeholder {
  color: #C6CDD4;
}

.c1:-webkit-autofill,
.c1:-webkit-autofill:hover,
.c1:-webkit-autofill:focus,
.c1:-webkit-autofill:active {
  -webkit-text-fill-color: #001E3E;
  -webkit-transition: background-color 99999999ms ease 99999999ms;
  transition: background-color 99999999ms ease 99999999ms;
}

.c1 + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
  color: #9CA7B4;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

.c1:disabled + .c2 {
  color: #C6CDD4;
}

.c1:-webkit-autofill + .c2,
.c1:-webkit-autofill:hover + .c2,
.c1:-webkit-autofill:focus + .c2,
.c1:-webkit-autofill:active + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
}

.c1:focus:not(:disabled) + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
  color: #096BDB;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

<div
  class="c0"
>
  <textarea
    class="sc-AxjAm sc-AxhCb c1"
    placeholder="FREE NOW"
  />
  <label
    class="sc-AxirZ c2 c3"
  >
    Name
  </label>
</div>
`;

exports[`Textarea renders with custom height and width 1`] = `
.c3 {
  position: absolute;
  pointer-events: none;
  background-color: transparent;
  line-height: 1.5;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  max-width: calc(100% - 2rem);
  -webkit-transition: top 100ms ease-out,left 100ms ease-out, padding 100ms ease-out,font-size 100ms ease-out, color 100ms ease-out,background 100ms ease-out;
  transition: top 100ms ease-out,left 100ms ease-out, padding 100ms ease-out,font-size 100ms ease-out, color 100ms ease-out,background 100ms ease-out;
  top: 0.75rem;
  left: 0.5rem;
  padding: 0 0.25rem;
  font-size: 1rem;
}

.c0 {
  display: inline-block;
  position: relative;
  box-sizing: border-box;
  width: 21.5rem;
  height: 7.25rem;
  width: 10rem;
  height: 10rem;
}

.c1 {
  margin: 0;
  box-sizing: border-box;
  background: #FFFFFF;
  border-radius: 0;
  color: #001E3E;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  -webkit-transition: box-shadow 100ms,border 100ms;
  transition: box-shadow 100ms,border 100ms;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 100%;
  border-radius: 0.25rem;
  border: 0.0625rem solid #C6CDD4;
  font-size: 1rem;
  height: 3rem;
  padding: 0 0.75rem;
  height: 100%;
  padding: 0.75rem;
  overflow: auto;
  resize: both;
}

.c1::-webkit-input-placeholder {
  color: #9CA7B4;
}

.c1::-moz-placeholder {
  color: #9CA7B4;
}

.c1:-ms-input-placeholder {
  color: #9CA7B4;
}

.c1::placeholder {
  color: #9CA7B4;
}

.c1:active,
.c1:focus {
  border-color: #096BDB;
  box-shadow: inset 0 0 0 0.0625rem #096BDB;
}

.c1:disabled {
  color: #C6CDD4;
  border-color: #C6CDD4;
  box-shadow: none;
  cursor: not-allowed;
}

.c1:disabled::-webkit-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::-moz-placeholder {
  color: #C6CDD4;
}

.c1:disabled:-ms-input-placeholder {
  color: #C6CDD4;
}

.c1:disabled::placeholder {
  color: #C6CDD4;
}

.c1:-webkit-autofill,
.c1:-webkit-autofill:hover,
.c1:-webkit-autofill:focus,
.c1:-webkit-autofill:active {
  -webkit-text-fill-color: #001E3E;
  -webkit-transition: background-color 99999999ms ease 99999999ms;
  transition: background-color 99999999ms ease 99999999ms;
}

.c1 + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
  color: #9CA7B4;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

.c1:disabled + .c2 {
  color: #C6CDD4;
}

.c1:-webkit-autofill + .c2,
.c1:-webkit-autofill:hover + .c2,
.c1:-webkit-autofill:focus + .c2,
.c1:-webkit-autofill:active + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
}

.c1:focus:not(:disabled) + .c2 {
  font-weight: 600;
  top: -0.5rem;
  font-size: 0.75rem;
  color: #096BDB;
  background: #FFFFFF;
  background: linear-gradient(0deg,#FFFFFF calc(50% + 0.0625rem),transparent 50%);
}

<div
  class="c0"
  height="10rem"
  width="10rem"
>
  <textarea
    class="sc-AxjAm sc-AxhCb c1"
    placeholder="FREE NOW"
  />
  <label
    class="sc-AxirZ c2 c3"
  >
    Name
  </label>
</div>
`;
